### PR TITLE
[FLINK-15792][k8s] Make Flink logs accessible via kubectl logs per default

### PR DESCRIPTION
**Pull request by wangyang0918**
## What is the purpose of the change

Currently, the users could not view the JobManager/TaskManager logs via `kubectl logs` by default. It is not convenient especially for debugging(i.e. when the entrypoint failed). So this PR suggests to make the logs output to console by default. This is also aligned with standalone session/job cluster.

Note: After this change, the STDOUT/STDERR will be only accessible via `kubectl logs`. It means we could not access them in the Flink web dashboard.

## Brief change log

* Use constant logging variable to replace constant strings
* Add log4j1 configuration to JobManager and TaskManager start command
* Make Flink logs accessible via kubectl logs per default
* Remove logging section in native Kubernetes document

## Verifying this change

* Covered by existing decorator tests
* The existing `kubernetes-application.sh` could cover this change. It greps the expected logs from console.
* Manually test on a K8s cluster

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (**yes** / no / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangyang0918** [comment]
cc @tillrohrmann Follow what we have discussed in FLINK-15792, this PR will make logs accessible via `kubectl logs` by default. Please have a look at your convenience.

**Comment by wangyang0918** [comment]
@flinkbot run azure

**Comment by wangyang0918** [comment]
cc @tillrohrmann, i have rebased the latest master and could you have a look at your convenience.

**Comment by tillrohrmann** [comment]
Thanks for rebasing the PR @wangyang0918. Me or someone from my team will take a look at it shortly.

File: docs/ops/deployment/native_kubernetes.md [hunk 1]
```diff
@@ -154,23 +154,10 @@ $ kubectl delete deployment/<ClusterID>
 
 ## Log Files
 
-By default, the JobManager and TaskManager only store logs under `/opt/flink/log` in each pod.
-If you want to use `kubectl logs <PodName>` to view the logs, you must perform the following:
+By default, the JobManager and TaskManager will output the logs to console and `/opt/flink/log` in each pod simultaneously.
```
Comment (rmetzger) [inline docs/ops/deployment/native_kubernetes.md:157]:
```suggestion
By default, the JobManager and TaskManager will output the logs to the console and `/opt/flink/log` in each pod simultaneously.
```

File: docs/ops/deployment/native_kubernetes.md [hunk 2]
```diff
@@ -154,23 +154,10 @@ $ kubectl delete deployment/<ClusterID>
...
+The STDOUT and STDERR will only be redirected to console. You could check for them via `kubectl logs <PodName>`.
```
Comment (rmetzger) [inline docs/ops/deployment/native_kubernetes.md:158]:
```suggestion
The STDOUT and STDERR will only be redirected to the console. You can access them via `kubectl logs <PodName>`.
```

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java [hunk 3]
```diff
@@ -213,10 +211,11 @@ private static String getLogging(String logFile, String confDir, boolean hasLogb
...
 			logging.append("-Dlog.file=").append(logFile);
 			if (hasLogback) {
-				logging.append(" -Dlogback.configurationFile=file:").append(confDir).append("/logback.xml");
+				logging.append(" -Dlogback.configurationFile=file:").append(confDir).append("/").append(CONFIG_FILE_LOGBACK_NAME);
 			}
 			if (hasLog4j) {
-				logging.append(" -Dlog4j.configurationFile=file:").append(confDir).append("/log4j.properties");
+				logging.append(" -Dlog4j.configuration=file:").append(confDir).append("/").append(CONFIG_FILE_LOG4J_NAME)
+					.append(" -Dlog4j.configurationFile=file:").append(confDir).append("/").append(CONFIG_FILE_LOG4J_NAME);
 			}
```
Comment (rmetzger) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java:219]: why is it necessary to define both `log4j.configuration` and `log4j.configurationFile` ?

File: flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/JavaCmdJobManagerDecoratorTest.java [hunk 4]
```diff
@@ -54,13 +56,14 @@
 
 	// Logging variables
 	private static final String logback =
-			String.format("-Dlogback.configurationFile=file:%s/logback.xml", FLINK_CONF_DIR_IN_POD);
-	private static final String log4j =
-			String.format("-Dlog4j.configurationFile=file:%s/log4j.properties", FLINK_CONF_DIR_IN_POD);
+			String.format("-Dlogback.configurationFile=file:%s/%s", FLINK_CONF_DIR_IN_POD, CONFIG_FILE_LOGBACK_NAME);
+	private static final String log4j = String.format(
+		"-Dlog4j.configuration=file:%s/%s -Dlog4j.configurationFile=file:%s/%s",
```
Comment (rmetzger) [inline flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/JavaCmdJobManagerDecoratorTest.java:61]: same question as above :)

**Comment by rmetzger** [comment]
While testing the change, I found that I can not access the TaskManager logs:
```
$ kubectl logs yolo-taskmanager-1-1
Start command : /bin/bash -c $JAVA_HOME/bin/java -classpath $FLINK_CLASSPATH -Xmx205520892 -Xms205520892 -XX:MaxDirectMemorySize=202165453 -XX:MaxMetaspaceSize=268435456 org.apache.flink.kubernetes.taskmanager.KubernetesTaskExecutorRunner -D taskmanager.memory.framework.off-heap.size=134217728b -D taskmanager.memory.network.max=67947725b -D taskmanager.memory.network.min=67947725b -D taskmanager.memory.framework.heap.size=134217728b -D taskmanager.memory.managed.size=271790903b -D taskmanager.cpu.cores=1.0 -D taskmanager.memory.task.heap.size=71303164b -D taskmanager.memory.task.off-heap.size=0b --configDir /opt/flink/conf -Dweb.tmpdir='/tmp/flink-web-dca91753-58da-4d06-90e3-31ae03e17931' 1> /opt/flink/log/taskmanager.out 2> /opt/flink/log/taskmanager.err
```

also via the JobManager Web UI, the TM logs are not accessible.

It seems that not .log file has been created. The taskmanger.err contains:
```
# more taskmanager.err
ERROR StatusLogger No Log4j 2 configuration file found. Using default configuration (logging only errors to the conso
le), or user programmatically provided configurations. Set system property 'log4j2.debug' to show Log4j 2 internal in
itialization logging. See https://logging.apache.org/log4j/2.x/manual/configuration.html for instructions on how to c
onfigure Log4j 2
```

**Comment by rmetzger** [comment]
How about adding a `kubectl logs` call to the bash e2e tests to ensure that all the logs are accessible?

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java (hunk 3 above)
Comment (wangyang0918) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java:219]: For standalone and Yarn deployment, we could support both log4j1 and log4j2. We should do this for K8s deployment. `log4j.configuration` is for log4j1 and `log4j.configurationFile` is for log4j2.

Comment (rmetzger) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java:219]: Thanks for the explanation! Makes sense.

**Comment by wangyang0918** [comment]
@rmetzger Thanks for your review and testing. Do have build a new image with this PR? After that, i think the logs could be accessed successfully. I have tested in a K8s cluster.

I have integrated your other comments via the two `fixup` commits, including log accessible checking via `kubectl logs` in the E2E test.

**Comment by wangyang0918** [comment]
@flinkbot run azure

**Comment by rmetzger** [comment]
> Do have build a new image with this PR? 

You are right. I didn't think enough when testing ;) I will test the PR again tomorrow.

**Comment by rmetzger** [comment]
I will merge this change now

**Pull request closed without merging**